controlplane/pkg/remote/proxy_network_service_server/proxy_network_service_server.go

```diff
@@ -146,6 +140,28 @@ func (srv *proxyNetworkServiceServer) updateParameters(ctx context.Context, requ
 	return localSrcIP, originalNetworkService
 }
 
+func (srv *proxyNetworkServiceServer) updateResponseLocal(response *connection.Connection, localSrcIP, destNsmName, originalNetworkService string) {
+	response.Mechanism.Parameters[vxlan.SrcIP] = localSrcIP
+	response.NetworkServiceManagers[1] = destNsmName
+	response.NetworkService = originalNetworkService
+}
+
+func (srv *proxyNetworkServiceServer) updateResponseRemote(ctx context.Context, response *connection.Connection, clusterInfoClient clusterinfo.ClusterInfoClient) {
+	remoteNodeIPConfiguration, err := clusterInfoClient.GetNodeIPConfiguration(ctx, &clusterinfo.NodeIPConfiguration{InternalIP: response.Mechanism.Parameters["dst_ip"]})
+	if err == nil {
```

Comment:
should we propagate an error if err != nil?

Reply:
If err != nil then we just do not need to change anything, error should be ignored